store/postgres/src/deployment_store.rs

```diff
@@ -311,27 +313,50 @@ impl DeploymentStore {
             }
         }
 
-        // apply modification groups
-        for (entity_type, entities) in inserts.iter_mut() {
-            count +=
-                self.insert_entities(entity_type, entities, conn, layout, ptr, &stopwatch)? as i32
+        // Apply modification groups:
+        // Each operation must respect the maximum number of bindings allowed in PostgreSQL queries,
+        // so we need to act in chunks whose size is defined by the number of entities times the
+        // number of attributes each entity type has.
+
+        // Inserts:
+        for (entity_type, mut entities) in inserts.into_iter() {
+            let chunk_size = {
+                let num_fields = (*entities[0].1).len();
+                POSTGRES_MAX_BINDINGS_PER_STATEMENT / num_fields
```

Comment:
If you did the chunking inside `Layout.insert_entities`, you know how many parameters each entity needs by looking at `Table.columns.len()` That doesn't account for `block_range` and `vid`, so you'd have to add 2 to that number to be on the safe side (though I don't think we ever write `vid`)